src/pybind/mgr/dashboard/frontend/src/app/core/navigation/navigation/navigation.component.ts

```diff
@@ -67,20 +62,28 @@ export class NavigationComponent implements OnInit, OnDestroy {
       this.multiClusterService.subscribe((resp: object) => {
         const clustersConfig = resp['config'];
         if (clustersConfig) {
-          this.clustersMap.clear();
+          const clusters: { key: string; value: MultiCluster }[] = [];
           Object.keys(clustersConfig).forEach((clusterKey: string) => {
             const clusterDetailsList = clustersConfig[clusterKey];
             clusterDetailsList.forEach((clusterDetails: MultiCluster) => {
               const clusterUser = clusterDetails['user'];
               const clusterUrl = clusterDetails['url'];
               const clusterUniqueKey = `${clusterUrl}-${clusterUser}`;
-              this.clustersMap.set(clusterUniqueKey, clusterDetails);
-              this.checkClusterConnectionStatus();
+              clusters.push({ key: clusterUniqueKey, value: clusterDetails });
             });
           });
+          clusters.sort((a, b) =>
+            a.value.cluster_alias === 'local-cluster'
+              ? -1
+              : b.value.cluster_alias === 'local-cluster'
+              ? 1
+              : 0
+          );
```

Comment:
this could be a shared function and should be available already maybe. If not please add this as a shared funtion and make it generic.

Reply:
Done